touchforms/formplayer/static/formplayer/script/entrycontrols_full.js

```diff
@@ -1,785 +1,584 @@
+};
 
-/* ============================= */
-
+InfoEntry.prototype = Object.create(Entry.prototype);
+InfoEntry.prototype.constructor = Entry;
 
-function Entry () {
-  this.help = function () {
-    showError(activeQuestion["help"] || "There is no help text for this question.");
-  }
-
-  this.clear = function () {
-    this.setAnswer(null, true);
-  }
-
-  this.onShow = function() { }
+/**
+ * The entry used when we have an unidentified entry
+ */
+function UnsupportedEntry(question, options) {
+    Entry.call(self, question, options);
+    this.templateType = 'unsupported'
+    this.answer = null;
 }
+UnsupportedEntry.prototype = Object.create(Entry.prototype);
+UnsupportedEntry.prototype.constructor = Entry;
 
-function SimpleEntry () {
-  inherit(this, new Entry());
+/**
+ * The entry that represents a free text input
+ * TODO: Handle prose option
+ */
+function FreeTextEntry(question, options) {
+    var self = this;
+    Entry.call(self, question, options);
+    self.templateType = 'str';
+    self.domain = question.domain ? question.domain() : 'full';
+    self.lengthLimit = options.lengthLimit || 10000;
+    self.prose = question.domain_meta ? question.domain_meta().longtext : false;
+
+    self.answer = question.answer;
+    self.answer.subscribe(function(newValue) {
+        self.question.onchange()
+    });
 
-  this.shortcuts = [];
+    this.prevalidate = function() {
+        var errmsg = this._prevalidate(self.answer());
+        if (errmsg) {
+            self.question.showError(errmsg);
+            return false;
+        }
+        return true;
+    }
 
-  this.prevalidate = function (q) {
-    return true;
-  }
+    this._prevalidate = function(raw) {
+        return null;
+    }
 
-  this.destroy = function () {
-    for (var i = 0; i < this.shortcuts.length; i++) {
-      shortcut.remove(this.shortcuts[i]);
+    this.domainText = function() {
+        return '(free-text)';
     }
-  }
+}
+FreeTextEntry.prototype = Object.create(Entry.prototype);
+FreeTextEntry.prototype.constructor = Entry;
 
-  this.add_shortcut = function (hotkey, func) {
-    set_shortcut(hotkey, func);
-    this.shortcuts.push(hotkey);
-  }
+function PasswordEntry(question, options) {
+    options.lengthLimit = options.lengthLimit || 9;
+
+    this.mkWidget = function() {
+        $('#answer')[0].innerHTML = '<input id="textfield" maxlength="' + this.lengthLimit + '" type="passwd"/>';
+        this.inputfield = $('#textfield')[0];
+    }
 }
+PasswordEntry.prototype = Object.create(FreeTextEntry.prototype);
+PasswordEntry.prototype.constructor = FreeTextEntry;
 
-function InfoEntry () {
-  inherit(this, new SimpleEntry());
-  this.answer = null;
+/**
+ * The entry that defines an integer input. Only accepts whole numbers
+ */
+function IntEntry(question, options) {
+    FreeTextEntry.call(this, question, options);
+    this.templateType = 'int';
+    this.lengthLimit = options.lengthLimit;
 
-  this.setAnswer = function (val) {
```

Comment:
we can completely do away with `setAnswer`, `getAnswer`, `load` and `mkWidget` since we're now using knockout which should take care of all that for us